dev/tests/api-functional/testsuite/Magento/GraphQl/ConfigurableProduct/AddConfigurableProductToCartSingleMutationTest.php

```diff
@@ -0,0 +1,292 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GraphQl\ConfigurableProduct;
+
+use Exception;
+use Magento\GraphQl\Quote\GetMaskedQuoteIdByReservedOrderId;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\TestCase\GraphQlAbstract;
+
+/**
+ * Add configurable product to cart testcases
+ */
+class AddConfigurableProductToCartSingleMutationTest extends GraphQlAbstract
+{
+    /**
+     * @var GetMaskedQuoteIdByReservedOrderId
+     */
+    private $getMaskedQuoteIdByReservedOrderId;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        $objectManager = Bootstrap::getObjectManager();
+        $this->getMaskedQuoteIdByReservedOrderId = $objectManager->get(GetMaskedQuoteIdByReservedOrderId::class);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/ConfigurableProduct/_files/product_configurable.php
+     * @magentoApiDataFixture Magento/Checkout/_files/active_quote.php
+     */
+    public function testAddConfigurableProductToCart()
+    {
+        $product = $this->getConfigurableProductInfo();
+        $quantity = 2;
+        $parentSku = $product['sku'];
+        $attributeId = (int) $product['configurable_options'][0]['attribute_id'];
+        $valueIndex = $product['configurable_options'][0]['values'][1]['value_index'];
+
+        $selectedConfigurableOptionsQuery = $this->generateSuperAttributesIdV2Query($attributeId, $valueIndex);
+        $maskedQuoteId = $this->getMaskedQuoteIdByReservedOrderId->execute('test_order_1');
+
+        $query = $this->getQuery(
+            $maskedQuoteId,
+            $product['sku'],
+            2,
+            $selectedConfigurableOptionsQuery
+        );
+
+        $response = $this->graphQlMutation($query);
+
+        $cartItem = current($response['addProductsToCart']['cart']['items']);
+        self::assertEquals($quantity, $cartItem['quantity']);
+        self::assertEquals($parentSku, $cartItem['product']['sku']);
+        self::assertArrayHasKey('configurable_options', $cartItem);
+
+        $option = current($cartItem['configurable_options']);
+        self::assertEquals($attributeId, $option['id']);
+        self::assertEquals($valueIndex, $option['value_id']);
+        self::assertArrayHasKey('option_label', $option);
+        self::assertArrayHasKey('value_label', $option);
+    }
+
+    /**
+     * Generates Id_v2 for super configurable product super attributes
```

Comment:
Hi @rogyar . Based on this arch PR https://github.com/magento/architecture/pull/396/files would be nice if you can update the id_v2 references to uid.